init.lua

```diff
@@ -29,50 +40,91 @@ castle_masonry.materials = {}
 end
-if read_setting("castle_masonry_cobble", true) then
-	table.insert(castle_masonry.materials, {name="cobble", desc=S("Cobble"), tile="default_cobble.png", craft_material="default:cobble"})
-end
-if read_setting("castle_masonry_stonebrick", true) then
-	table.insert(castle_masonry.materials, {name="stonebrick", desc=S("Stonebrick"), tile="default_stone_brick.png", craft_material="default:stonebrick"})
-end
-if read_setting("castle_masonry_sandstonebrick", true) then
-	table.insert(castle_masonry.materials, {name="sandstonebrick", desc=S("Sandstone Brick"), tile="default_sandstone_brick.png", craft_material="default:sandstonebrick"})
-end
-if read_setting("castle_masonry_desertstonebrick", true) then
-	table.insert(castle_masonry.materials, {name="desertstonebrick", desc=S("Desert Stone Brick"), tile="default_desert_stone_brick.png", craft_material="default:desert_stonebrick"})
-end
-if read_setting("castle_masonry_desertsandstonebrick", true) then
-	table.insert(castle_masonry.materials, {name="desertsandstonebrick", desc=S("Desert Sandstone Brick"), tile="default_desert_sandstone_brick.png", craft_material="default:desert_sandstone_brick"})
-end
-if read_setting("castle_masonry_silversandstonebrick", true) then
-	table.insert(castle_masonry.materials, {name="silversandstonebrick", desc=S("Silver Sandstone Brick"), tile="default_silver_sandstone_brick.png", craft_material="default:silver_sandstone_brick"})
-end
-if read_setting("castle_masonry_stone", true) then
-	table.insert(castle_masonry.materials, {name="stone", desc=S("Stone"), tile="default_stone.png", craft_material="default:stone"})
-end
-if read_setting("castle_masonry_sandstone", true) then
-	table.insert(castle_masonry.materials, {name="sandstone", desc=S("Sandstone"), tile="default_sandstone.png", craft_material="default:sandstone"})
-end
-if read_setting("castle_masonry_desertstone", true) then
-	table.insert(castle_masonry.materials, {name="desertstone", desc=S("Desert Stone"), tile="default_desert_stone.png", craft_material="default:desert_stone"})
-end
-if read_setting("castle_masonry_desertsandstone", true) then
-	table.insert(castle_masonry.materials, {name="desertsandstone", desc=S("Desert Sandstone"), tile="default_desert_sandstone.png", craft_material="default:desert_sandstone"})
-end
-if read_setting("castle_masonry_silversandstone", true) then
-	table.insert(castle_masonry.materials, {name="silversandstone", desc=S("Silver Sandstone"), tile="default_silver_sandstone.png", craft_material="default:silver_sandstone"})
-end
-if read_setting("castle_masonry_wood", false) then
-	table.insert(castle_masonry.materials, {name="wood", desc=S("Wood"), tile="default_wood.png", craft_material="group:wood", composition_material="default:wood"})
+if minetest.get_modpath("default") then
+	if read_setting("castle_masonry_cobble", true) then
```

Comment:
The previous code was very repetitive and this change does not improve the situation either.

What would you think of using a helper function to further reduce code redundancy? (not tested!)

```Lua
local function register_material(setting_default, def)
	if read_setting("castle_masonry_" + def.name, setting_default) then
		table.insert(castle_masonry.materials, def)
	end
end


-- Example
register_material({name="sandstonebrick", desc=S("Sandstone Brick"), tile="default_sandstone_brick.png", craft_material="default:sandstonebrick"})
register_material({name="desertstonebrick", desc=S("Desert Stone Brick"), tile="default_desert_stone_brick.png", craft_material="default:desert_stonebrick"})
```

Reply:
done